client.cpp

```diff
@@ -22,482 +22,476 @@
+    WSADATA wsaData;
+    int iResult;
+    char recvbuf[DEFAULT_BUFLEN + MSG_HEADER];
+    int recvbuflen = DEFAULT_BUFLEN + MSG_HEADER;
+    int rtnVal;
+    
+    iResult = WSAStartup(MAKEWORD(2,2), &wsaData);
+    if(iResult != 0){
+        printf("WSAStartup failed: %d\n", iResult);
+        exit(1);
+    }
+    
+    struct sockaddr_in servAddr;
+    ZeroMemory(&servAddr, sizeof(servAddr));
+    servAddr.sin_family = AF_INET;
 
-	const char* theIp = params.ipAddr.c_str();// = "127.0.0.1"; // pass in ip as arg to client?
-	
-	int size = sizeof(servAddr);
-	iResult = WSAStringToAddressA((LPSTR)theIp, AF_INET, NULL, (struct sockaddr *)&servAddr, &size);
-	if(iResult != 0){
-		printf("getaddrinfo failed: %d\n", iResult);
-		WSACleanup();
-	}else{
-		servAddr.sin_port = htons(DEFAULT_PORT);
-	}
-	
-	SOCKET connectSocket = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
-	if(connectSocket == INVALID_SOCKET){
-		printf("socket() failed\n");
-		WSACleanup();
-		exit(1);
-	}
-	
-	// maybe move all this stuff to a separate thread?
-	// then in this main thread we can handle some basic SDL2 window stuff, like closing out the window
-	printf("preparing to message server...\n");
-	std::string str = "hellooo there"; 
-	const char* msg = str.c_str();
-	
-	bool sentInitMsg = false;
-	
-	int packetsToExpect = 0;
-	int packetsReceived = 0;
-	int lastFrameId = -1;
-	int packetNum;
-	int currFrameId;
-	
-	// we can cheat a little for now. we know exactly how big the image is coming in,
-	// so we know exactly how many pixel data bytes can go into each packet. 
-	// for now, let's use this knowledge (i.e. there should be 16 packets, each one with 
-	// 60000 bytes of data except the last packet, which has 50000). this is beacause we hardcoded
-	// the width and height of the image, which is 600 and 400, respectively.
-	int dataSize = DEFAULT_BUFLEN; // this is specific to the current parameters (image coming in total size is 960000 bytes)
-	SDL_Texture* texBuf = nullptr;
-	
-	while(true){
-		
-		// set up time stuff for select timeout
-		// https://stackoverflow.com/questions/21743231/winsock-selects-timeout-on-listening-socket-causing-every-subsequent-select-c
-		timeval timeout;
-		timeout.tv_sec = 2;
-		timeout.tv_usec = 0;
-		fd_set fds;
-		FD_ZERO(&fds);
-		FD_SET(connectSocket, &fds);
-			
-		// clear the recv buffer 
-		ZeroMemory(recvbuf, recvbuflen);
-		
-		// send msg 
-		if(!sentInitMsg){
-			int bytesSent = sendto(connectSocket, msg, str.length(), 0, (struct sockaddr *)&servAddr, size);
-			if(bytesSent < 0){
-				printf("sendto failed.\n");
-				exit(1);
-			}
-			std::cout << "sent hello msg" << std::endl;
-			sentInitMsg = true;
-		}
-		
-		// receive msg
-		if(select(0, &fds, 0, 0, &timeout) == 0){
-			// we got a timeout
-			std::cout << "socket recvfrom timed out. process whatever data is in the priority queue." << std::endl;
+    const char* theIp = params.ipAddr.c_str();
+    servAddr.sin_addr.s_addr = inet_addr(theIp);
```

Comment:
I think the root of the issue was around here. using WSAStringToAddressA didn't seem to work (but not really sure why I didn't see any issues with using 127.0.0.1)